### ISSUE TITLE
Add Response.ResponseType.SUCCESS_ATOM_FEED

### ISSUE THREAD

**Issue by neumino** [rethinkdb/rethinkdb#3389]
Any chance to introduce a new term like `Response.ResponseType.SUCCESS_ATOM_FEED` such that libraries know if the query returned a feed for a table/stream, or for a unique value/document?

The driver could treat a SUCCESS_FEED and a SUCCESS_ATOM_FEED the same way, except that there would be a new method like `feed.isAtom()`.

The reason behind is that I would like to have the following interface in thinky.

```js
var Channel = thinky.createModel("Channel", { /* schema */ })
var echo = sockjs.createServer({ sockjs_url: 'http://cdn.jsdelivr.net/sockjs/0.3.4/sockjs.min.js' });
echo.on('connection', function(conn) {
    var channel;
    conn.on('data', function(id) {
        if (channel === undefined) {
            channel = null;  
            Channel.get(id).changes().run().then(function(channel) { // channel is an instance of Channel here
                conn.write(channel.toJSON());
                // I would prefer not to create an instance of Channel everytime, but keep the same object and just update it.
                channel.on("change", function() {
                    conn.write(channel.toJSON());
                })
            })
        }
    });
    conn.on('close', function() {
        // do some clean up
    });
});
```

Basically for a point changefeed, I would like to return not a feed, but an actual object that automatically updates itself and emit events (thinky will wrap the feed).
The thing is now I have no way to know if I'll get back a stream of a single value/object.

I could work around by removing `changes` and adding an option in `run`, but I would prefer to stick close to ReQL if possible.

**Comment by gchpaco** [comment]
CR 2487.

**Comment by deontologician** [comment]
Now that point changefeeds are going to use the regular format, is this
still needed?

**Comment by danielmewes** [comment]
It still seems to be useful for @neumino original goal.

**Comment by neumino** [comment]
What is the regular format for feeds?

**Comment by deontologician** [comment]
`{new_val: ... , old_val: ...}` see #3564

**Comment by neumino** [comment]
Oh ok. Yea I would still find this type useful as I don't really want to wait for the first update to know what kind of feed I have.

**Comment by deontologician** [comment]
Ah ok, that makes sense
